src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.Windows.cs

```diff
@@ -0,0 +1,180 @@
+            var serverStrean = new FileStream(new SafeFileHandle(server.SafePipeHandle.DangerousGetHandle(), true), FileAccess.Read);
+
+            server.SafePipeHandle.SetHandleAsInvalid();
+
+            return (serverStrean, clienStream);
+        }
+
+        protected override Type UnsupportedConcurrentExceptionType => null;
+        protected override bool UsableAfterCanceledReads => false;
+        protected override bool FullyCancelableOperations => false;
+        protected override bool BlocksOnZeroByteReads => OperatingSystem.IsWindows();
+        protected override bool SupportsConcurrentBidirectionalUse => false;
+    }
+
+    [PlatformSpecific(TestPlatforms.Windows)] // DOS device paths (\\.\ and \\?\) are a Windows concept
+    public class SeekableDeviceFileStreamStandaloneConformanceTests : UnbufferedAsyncFileStreamStandaloneConformanceTests
+    {
+        protected override string GetTestFilePath(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0)
+        {
+            string filePath = Path.GetFullPath(base.GetTestFilePath(index, memberName, lineNumber));
+            string drive = Path.GetPathRoot(filePath);
+            StringBuilder volumeNameBuffer = new StringBuilder(filePath.Length + 1024);
+
+            // the following method maps drive letter like "C:\" to a DeviceID (a DOS device path)
+            // example: "\\?\Volume{724edb31-eaa5-4728-a4e3-f2474fd34ae2}\"
+            if (!GetVolumeNameForVolumeMountPoint(drive, volumeNameBuffer, volumeNameBuffer.Capacity))
+            {
+                throw new Win32Exception(Marshal.GetLastPInvokeError(), "GetVolumeNameForVolumeMountPoint failed");
+            }
+
+            // instead of:
+            // 'C:\Users\x\AppData\Local\Temp\y\z
+            // we want sth like:
+            // '\\.\Volume{724edb31-eaa5-4728-a4e3-f2474fd34ae2}\Users\x\AppData\Local\Temp\y\z
+            string devicePath = filePath.Replace(drive, volumeNameBuffer.ToString());
+            Assert.StartsWith(@"\\?\", devicePath);
+#if DEBUG
+            // we do want to test \\.\ prefix as well
+            devicePath = devicePath.Replace(@"\\?\", @"\\.\");
+#endif
+
+            return devicePath;
+        }
+
+        [DllImport(Interop.Libraries.Kernel32, EntryPoint = "GetVolumeNameForVolumeMountPointW", CharSet = CharSet.Unicode, BestFitMapping = false, SetLastError = true)]
+        private static extern bool GetVolumeNameForVolumeMountPoint(string volumeName, StringBuilder uniqueVolumeName, int uniqueNameBufferCapacity);
+    }
+
+    [PlatformSpecific(TestPlatforms.Windows)] // the test setup is Windows-specifc
+    [Collection("NoParallelTests")] // don't run in parallel, as file sharing logic is not thread-safe
+    [OuterLoop("Requires admin privileges to create a file share")]
+    [ConditionalClass(typeof(UncFilePathFileStreamStandaloneConformanceTests), nameof(UncFilePathFileStreamStandaloneConformanceTests.CanShareFiles))]
```

Comment:
Do you know whether the condition ends up being evaluated per test or per class? I don't know the answer,  but if it's the former, might want to cache the static property... it looks pretty expensive.